cmd/soroban-rpc/internal/db/mocks.go

```diff
@@ -72,12 +76,31 @@ func (txn *mockTransactionHandler) GetTransaction(ctx context.Context, hash xdr.
 	if tx, ok := txn.txs[hash.HexString()]; !ok {
 		return Transaction{}, txn.ledgerRange, ErrNoTransaction
 	} else {
-		itx, err := ParseTransaction(*txn.ledgers[hash.HexString()], tx)
+		itx, err := ParseTransaction(*txn.txHashToMeta[hash.HexString()], tx)
 		return itx, txn.ledgerRange, err
 	}
 }
 
 func (txn *mockTransactionHandler) RegisterMetrics(_, _ prometheus.Observer) {}
 
-var _ TransactionReader = &mockTransactionHandler{}
-var _ TransactionWriter = &mockTransactionHandler{}
```

Comment:
why remove? :confused:

Reply:
in fact you should instead add a line ensuring `mockLedgerReader` conforms to the `db.LedgerReader` interface

Reply:
@Shaptic Ah I think I did not realise that these lines are actually needed so that it gives an error if the structs stop implementing the interfaces. I will add them again and also the mockLedgerReader